Improve code unicorn

rtritto opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/mongo-express/mongo-express/994)
- - -
<!-- Reviewable:end -->
Improve code using [eslint-plugin-unicorn](https://www.npmjs.com/package/eslint-plugin-unicorn).

## 1) Fixed rules:
- [prefer-number-properties](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-number-properties.md)
- [prefer-string-slice](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-string-slice.md)
- [no-array-for-each](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-array-for-each.md)
- [no-for-loop](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-for-loop.md)
- [explicit-length-check](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/explicit-length-check.md)
- [no-await-expression-member](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-await-expression-member.md)
- [prefer-top-level-await](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-top-level-await.md)
- [empty-brace-spaces](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/empty-brace-spaces.md)
- [consistent-function-scoping](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/consistent-function-scoping.md)
- [prefer-node-protocol](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-node-protocol.md)
- [prefer-includes](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-includes.md)
- [prefer-set-has](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-set-has.md)
- [prefer-spread](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-spread.md)
- [prefer-array-flat](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-array-flat-map.md)
- [prefer-query-selector](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-query-selector.md)
- [better-regex](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/better-regex.md)
- [no-lonely-if](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-lonely-if.md)
- [prefer-at](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-at.md)
- [prefer-optional-catch-binding](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-optional-catch-binding.md)
- [prefer-native-coercion-functions](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-native-coercion-functions.md)
  - Added eslint-disable-next-line in file `/lib/routes/collection.js`
- [consistent-destructuring](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/consistent-destructuring.md)
- [no-nested-ternary](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-nested-ternary.md)
- [prefer-regexp-test](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-regexp-test.md)
- [catch-error-name](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/catch-error-name.md)

## 2) Rules where help is needed (discuss if add to ignore):
- [prefer-export-from](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-export-from.md)
  - files: `/lib/scripts/codeMirrorLoader.js`, `/lib/scripts/jQueryLoader.js`
- [no-unsafe-regex](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-unsafe-regex.md)
  - file: `/lib/filter.js`
- [numeric-separators-style](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/numeric-separators-style.md)
  - files: `/ib/router.js`, `/test/testMongoConfig.js`, `webpack.config.js`
- [no-array-callback-reference](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-array-callback-reference.md)
  - file: `/lib/routes/collection.js`
- [no-array-method-this-argument](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-array-method-this-argument.md)
  - file: `/lib/routes/collection.js`

## 3) Ignored rules or TBD:
- [prevent-abbreviations](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prevent-abbreviations.md)
- [filename-case](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/filename-case.md)
- [no-keyword-prefix](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-keyword-prefix.md)
- [no-null](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-null.md)
- [prefer-string-replace-all](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-string-replace-all.md)
  - Added to ignore because of missing replaceAll function in Node v14

shakaran commented:
@rtritto probably if you gonna apply a lot rules in batch, maybe it is better to do a PR per rule when each one is gonna merge, link here. The full changes are already pretty big if we have to revert some specific rule.

If you want I can merge and squash the current status and you can make separate PR linking here for the new rules

rtritto commented:
I'm doing 1 commit per rule.
Next we should use merge instead of squash. By this way a single commit can be reverted and there will be less conflicts/diff.

Some rules should be discuss (this is an initial step).

rtritto commented on lib/scripts/collection.js:
```diff
@@ -4,20 +4,20 @@
 
 function getParameterByName(name) {
-  name = name.replace(/\[/, '\\[').replace(/[\]]/, '\\]');
+  name = name.replace(/\[/, '\\[').replace(/]/, '\\]');
```
```console
/[\]]/ can be optimized to /]/                        unicorn/better-regex
```

This fix triggers CodeQL.

@BlackthornYugen @shakaran

rtritto commented:
@shakaran rebased.
I have to do other rule fixes. Meantime you can try and/or check `2)` and `3)`.

shakaran commented:
> @shakaran rebased. I have to do other rule fixes. Meantime you can try and/or check `2)` and `3)`.

For

[prefer-export-from](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-export-from.md)
files: /lib/scripts/codeMirrorLoader.js, /lib/scripts/jQueryLoader.js => YES, FOR
[no-unsafe-regex](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-unsafe-regex.md)
file: /lib/filter.js => YES, FOR
[numeric-separators-style](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/numeric-separators-style.md)
files: /ib/router.js, /test/testMongoConfig.js, webpack.config.js => NO, AGAINST

rtritto commented:
> For
> 
> [prefer-export-from](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-export-from.md) files: /lib/scripts/codeMirrorLoader.js, /lib/scripts/jQueryLoader.js => YES, FOR [no-unsafe-regex](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-unsafe-regex.md) file: /lib/filter.js => YES, FOR [numeric-separators-style](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/numeric-separators-style.md) files: /ib/router.js, /test/testMongoConfig.js, webpack.config.js => NO, AGAINST

What do you mean? Do you have some fix?

shakaran commented:
> > For
> > [prefer-export-from](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-export-from.md) files: /lib/scripts/codeMirrorLoader.js, /lib/scripts/jQueryLoader.js => YES, FOR [no-unsafe-regex](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-unsafe-regex.md) file: /lib/filter.js => YES, FOR [numeric-separators-style](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/numeric-separators-style.md) files: /ib/router.js, /test/testMongoConfig.js, webpack.config.js => NO, AGAINST
> 
> What do you mean? Do you have some fixes?

For that cases, I said if I like or not the rule to apply

rtritto commented:
> For that cases, I said if I like or not the rule to apply

Ok, so we need to find the 2 fixes for _prefer-export-from_ and _no-unsafe-regex_.

There is another [open point](https://github.com/mongo-express/mongo-express/pull/994#discussion_r980984480).

rtritto commented:
@shakaran I finished all my possible fixes.
Added `no-array-callback-reference`, `no-array-method-this-argumen` and `no-null` and they will need to discuss.
Marked this [open point](https://github.com/mongo-express/mongo-express/pull/994#discussion_r980984480) as false positive (?) because cause unit tests fail: added it to ignore.

FYI @BlackthornYugen
